src/bluez/adapter/peripheral.rs

```diff
@@ -311,6 +316,26 @@ impl Peripheral {
                 .collect();
         }
 
+        if let Some(service_data) = args.service_data() {
+            properties.service_data = service_data
+                .into_iter()
+                .filter_map(|(service, data)| {
+                    let service: Uuid = service.parse().unwrap();
+                    if let Some(data) = cast::<Vec<u8>>(&data.0) {
+                        self.adapter.emit(CentralEvent::ServiceAdvertisement {
+                            address: self.address(),
+                            service,
+                            data: data.clone(),
+                        });
+                        Some((service, data.to_owned()))
+                    } else {
+                        warn!("Service data had wrong type: {:?}", &data.0);
+                        None
+                    }
+                })
+                .collect();
+        }
+
```

Comment:
Ditto here, let's only send the `DeviceUpdated` event if some property other than service/manufacturer data changed.